utils/fs/memory/memory.go

```diff
@@ -323,11 +311,35 @@ func (c *content) ReadAt(b []byte, off int64) (int, error) {
 		l = c.size - off
 	}
 
-	n := copy(b, c.bytes[off:l])
+	n := copy(b, c.bytes[off:off+l])
 	return n, nil
 }
 
 func (c *content) Truncate() {
-	c.bytes = []byte{}
+	c.bytes = make([]byte, 0)
```

Comment:
Do we really want to throw away the preallocated buffer when truncating, we are wasting resources here.